rs/cli/src/registry_dump.rs

```diff
@@ -0,0 +1,161 @@
+    DATA_CENTER_KEY_PREFIX, NODE_OPERATOR_RECORD_KEY_PREFIX, NODE_RECORD_KEY_PREFIX, SUBNET_RECORD_KEY_PREFIX,
+};
+
+use ic_protobuf::registry::{
+    dc::v1::DataCenterRecord, node::v1::NodeRecord, node_operator::v1::NodeOperatorRecord, subnet::v1::SubnetRecord,
+};
+
+use anyhow::Error;
+use ic_interfaces_registry::{RegistryClient, RegistryValue};
+use ic_management_types::Network;
+use ic_registry_local_registry::LocalRegistry;
+use service_discovery::registry_sync::sync_local_registry;
+use slog::Logger;
+use uuid::Uuid;
+
+pub async fn dump_registry(
+    logger: Logger,
+    path: &Option<PathBuf>,
+    version: &i64,
+    network: &Network,
+) -> Result<(), Error> {
+    let (path, should_dispose) = match path {
+        Some(val) => (val.clone(), false),
+        None => {
+            let uuid = Uuid::new_v4();
+            let binding = format!("~/tmp/{}", uuid);
+            let local_temp = shellexpand::tilde(&binding);
+            (
+                PathBuf::from_str(&local_temp).map_err(|e| anyhow::anyhow!("Error converting to path: {:?}", e))?,
+                true,
+            )
+        }
+    };
+
+    sync_local_registry(logger.clone(), path.clone(), vec![network.get_url()], false, None, None)
+        .await
+        .map_err(|e| anyhow::anyhow!("Error while syncing registry: {:?}", e))?;
+
+    // nodes,dc,node_operators,node_providers,
+    let client = LocalRegistry::new(path.clone(), Duration::from_secs(10))
+        .map_err(|e| anyhow::anyhow!("Couldn't create local registry instance: {:?}", e))?;
+
+    // determine desired version
+    let version = {
+        if *version >= 0 {
+            *version as u64
+        } else {
+            client.get_latest_version().get()
+        }
+    };
+    let version = RegistryVersion::new(version);
+
+    let nodes = client
+        .get_family_entries_versioned::<NodeRecord>(version)
+        .map_err(|e| anyhow::anyhow!("Couldn't get data centers: {:?}", e))?;
+    let nodes = serde_json::to_string(&nodes).map_err(|e| anyhow::anyhow!("Couldn't convert to JSON: {:?}", e))?;
+
+    let subnets = client
+        .get_family_entries_versioned::<SubnetRecord>(version)
+        .map_err(|e| anyhow::anyhow!("Couldn't get data centers: {:?}", e))?;
+    let subnets = serde_json::to_string(&subnets).map_err(|e| anyhow::anyhow!("Couldn't convert to JSON: {:?}", e))?;
+
+    let dcs = client
+        .get_family_entries_versioned::<DataCenterRecord>(version)
+        .map_err(|e| anyhow::anyhow!("Couldn't get data centers: {:?}", e))?;
+    let dcs = serde_json::to_string(&dcs).map_err(|e| anyhow::anyhow!("Couldn't convert to JSON: {:?}", e))?;
+
+    let node_operators = client
+        .get_family_entries_versioned::<NodeOperatorRecord>(version)
+        .map_err(|e| anyhow::anyhow!("Couldn't get data centers: {:?}", e))?;
+    let node_operators =
+        serde_json::to_string(&node_operators).map_err(|e| anyhow::anyhow!("Couldn't convert to JSON: {:?}", e))?;
+
```

Comment:
There is also node providers and node rewards that come to my mind.

Reply:
I will close this PR since if we integrate the backend code we don't need serilog yet. So to not complicate it I will migrate